unittests/backends/quantinuum/QuantinuumTester.cpp

```diff
@@ -289,6 +289,131 @@ CUDAQ_TEST(QuantinuumTester, checkObserveAsyncLoadFromFile) {
   EXPECT_TRUE(isValidExpVal(result.expectation()));
 }
 
+CUDAQ_TEST(QuantinuumTester, checkControlledRotations) {
+  // Checks for more advanced controlled rotations that only
+  // work in emulation.
+  std::string home = std::getenv("HOME");
+  std::string fileName = home + "/FakeCppQuantinuum.config";
+  auto backendString =
+      fmt::format(fmt::runtime(backendStringTemplate), mockPort, fileName);
+
+  auto &platform = cudaq::get_platform();
+  platform.setTargetBackend(backendString);
+
+  // rx: pi
+  {
+    auto kernel = cudaq::make_kernel();
+    auto controls1 = kernel.qalloc(2);
+    auto controls2 = kernel.qalloc(2);
+    auto control3 = kernel.qalloc();
+    auto target = kernel.qalloc();
+
+    // All of our controls in the 1-state.
+    kernel.x(controls1);
+    kernel.x(controls2);
+    kernel.x(control3);
+
+    kernel.rx<cudaq::ctrl>(M_PI, controls1, controls2, control3, target);
+
+    std::cout << kernel.to_quake() << "\n";
+
+    auto counts = cudaq::sample(kernel);
+    counts.dump();
+
+    // Target qubit should've been rotated to |1>.
+    EXPECT_EQ(counts.count("0000111111"), 1000);
```

Comment:
@anthony-santana why do these checks have more bits than qubits allocated?

Reply:
Oh nevermind, this is because of the ancilla bits being added by the multi-control-decomp